src/plugins/BlockWithMenu/Components.tsx

```diff
@@ -0,0 +1,162 @@
+  return (
+    <ul role="menubar" className="block-menu">
+      {!!props.children &&
+        React.Children.map(props.children, (child) => (
+          <li role="menuitem" key={child.props.title}>
+            {child}
+          </li>
+        ))}
+      <li role="menuitem">
+        <ToggleButton
+          value={spoilers}
+          title="Toggle Spoilers"
+          onValueChange={(spoilers) =>
+            props.updateAttributes?.({
+              spoilers,
+            })
+          }
+        >
+          {spoilers ? <EyeAlt /> : <EyeOff />}
+        </ToggleButton>
+      </li>
+      <li role="menuitem">
+        <Button
+          title={`Delete ${props.deleteTitle}`}
+          onClick={() => props.deleteNode?.()}
+        >
+          <Trash />
+        </Button>
+      </li>
+      <li role="menuitem">
+        <Button
+          title="Insert Paragraph Above"
+          onClick={() => {
+            if (props.getPos) {
+              props.editor
+                ?.chain()
+                .insertContentAt(props.getPos(), "<p></p>", {
+                  updateSelection: true,
+                })
+                .focus()
+                .run();
+            }
+          }}
+        >
+          Insert Paragraph <ArrowUp />
+        </Button>
+      </li>
+      <li role="menuitem">
+        <Button
+          title="Insert Paragraph Below"
+          onClick={() => {
+            if (props.getPos) {
+              props.editor
+                ?.chain()
+                .insertContentAt(props.getPos() + 1, "<p></p>", {
+                  updateSelection: true,
+                })
+                .focus()
+                .run();
+            }
+          }}
+        >
+          Insert Paragraph <ArrowDown />
+        </Button>
+      </li>
+    </ul>
+  );
+};
+
+export interface BlockBaseProps {
+  pluginName: string;
+  attributes: Attrs;
+  children?: React.ReactNode;
+  enclosingTag?: React.ElementType;
+  editable?: boolean;
+  className?: string;
+}
+
+export const BlockBaseComponent = (props: BlockBaseProps) => {
+  const attributes = props.attributes;
+  const Tag = props.enclosingTag ?? "div";
+  const dataAttributes = makeDataAttributes(attributes);
+  return (
+    <Tag
+      className={props.className ?? "base-block"}
+      tabIndex={props.editable || !attributes.spoilers ? -1 : 0}
+      data-type={props.pluginName}
+      {...dataAttributes}
+      style={{
+        width: attributes.width,
+        height: attributes.height,
+        display: "block",
+        maxWidth: "100%",
+      }}
+    >
+      {props.children}
+    </Tag>
+  );
+};
+
+export const EditableBlockWithMenuComponent = (
+  props: Partial<NodeViewProps> &
+    Required<Pick<NodeViewProps, "node" | "editor">>
+) => {
+  const attributes = props.node.attrs as BlockWithMenuOptions;
+  return (
+    <NodeViewWrapper data-type={PLUGIN_NAME}>
+      <BlockBaseMenu {...props} deleteTitle="block">
+        <Button
```

Comment:
What do we need these buttons for? Was trying to find them in the storybook preview but don't see them.

Reply:
These buttons are really just there for testing the BlockWithMenu plugin on it's own, before extending it. They won't show up in any of the child extensions. They can be seen in the Block With Menu story: https://deploy-preview-25--boba-editor-next.netlify.app/?path=/story/editor--block-with-menu
Note that the first one doesn't have data-width and data-height set in the initial content, so you only see the menu when it loads, but you can set them with the button and see it show up